src/simplification/SimplifyProperties.ts

```diff
@@ -16,56 +16,71 @@ export default function simplifyProperties(schema: Schema | boolean, simplifier
-      for (const [prop, propSchema] of Object.entries(schema.properties)) {
-        const newModels = simplifier.simplify(propSchema);
-        if (newModels.length > 0) {
-          addToProperty(prop, newModels[0]);
-        }
+  
+    for (const [prop, propSchema] of Object.entries(schema.properties || {})) {
+      const newModels = simplifier.simplify(propSchema);
+      if (newModels.length > 0) {
+        addToProperty(prop, newModels[0], schema, output);
       }
     }
+
     //If we encounter combination schemas ensure we recursively find the properties
     if (simplifier.options.allowInheritance !== true) {
       //Only merge allOf schemas if we don't allow inheritance
-      handleCombinationSchemas(schema.allOf);
+      combineSchemas(schema.allOf, output, simplifier, seenSchemas, schema);
     }
-    handleCombinationSchemas(schema.oneOf);
-    handleCombinationSchemas(schema.anyOf);
+    combineSchemas(schema.oneOf, output, simplifier, seenSchemas, schema);
+    combineSchemas(schema.anyOf, output, simplifier, seenSchemas, schema);
 
     //If we encounter combination schemas ensure we recursively find the properties
-    if (schema.then) {
-      addProperties(simplifyProperties(schema.then, simplifier, seenSchemas));
-    }
-    if (schema.else) {
-      addProperties(simplifyProperties(schema.else, simplifier, seenSchemas));
-    }
-    if (Object.keys(output).length === 0) {
-      return undefined;
-    }
-    return output;
+    combineSchemas(schema.then, output, simplifier, seenSchemas, schema);
+    combineSchemas(schema.else, output, simplifier, seenSchemas, schema);
+
+    return !Object.keys(output).length ? undefined : output;
   }
   return undefined;
+}
+
+/**
+ * Adds a property to the model
+ * 
+ * @param propName name of the property
+ * @param propModel the corresponding model
+ * @param schema the schema for the model
+ * @param currentModel the current output
+ */
+function addToProperty(propName: string, propModel: CommonModel, schema: Schema, currentModel: Output) {
+  if (currentModel === undefined) return;
+  //If a simplified property already exist, merge the two
+  if (currentModel[`${propName}`] !== undefined) {
+    currentModel[`${propName}`] = CommonModel.mergeCommonModels(currentModel[`${propName}`], propModel, schema);
+  } else {
+    currentModel[`${propName}`] = propModel;
+  }
+}
+
+/**
+ * Go through schema(s) and combine the simplified properties together.
+ * 
+ * @param schema to go through
+ * @param currentModel the current output
+ * @param simplifier the simplifier to use
+ * @param seenSchemas schemas which we already have outputs for
+ * @param rootSchema the root schema we are combining schemas for
+ */
+function combineSchemas(schema: (Schema | boolean) | (Schema | boolean)[] | undefined, currentModel: Output, simplifier : Simplifier, seenSchemas: Map<any, Output>, rootSchema : Schema) {
+  if (schema === undefined) return;
+  if (typeof schema === 'boolean') return;
```

Comment:
```suggestion
  if (typeof schema !== 'object') return;
```

Reply:
We have to check for undefined because it can be that type. It is easier in the main functionality to allow such value in this sub-function.

Reply:
As I wrote in another PR. For undefined you have also `false`.